webhooks/mutating_admission_webhook.go

```diff
@@ -42,66 +42,107 @@ func (m *PodMutator) Handle(ctx context.Context, req admission.Request) admissio
 			return admission.Allowed("openfeature is disabled")
 		}
 	}
-	// Check if the pod is static or orphaned
-	name := pod.Name
-	if len(pod.GetOwnerReferences()) != 0 {
-		name = pod.GetOwnerReferences()[0].Name
-	} else {
-		return admission.Denied("static or orphaned pods cannot be mutated")
-	}
 
-	var featureFlagCustomResource corev1alpha1.FeatureFlagConfiguration
-	// Check CustomResource
+	// Check configuration
 	val, ok = pod.GetAnnotations()["openfeature.dev/featureflagconfiguration"]
 	if !ok {
 		return admission.Allowed("FeatureFlagConfiguration not found")
-	} else {
-		// Current limitation is to use the same namespace, this is easy to fix though
-		// e.g. namespace/name check
-		err = m.Client.Get(context.TODO(), client.ObjectKey{Name: val,
-			Namespace: req.Namespace},
-			&featureFlagCustomResource)
+	}
+
+	// Check if the pod is static or orphaned
+	if len(pod.GetOwnerReferences()) == 0 {
+		return admission.Denied("static or orphaned pods cannot be mutated")
+	}
+
+	// Check for ConfigMap and create it if it doesn't exist
+	cm := corev1.ConfigMap{}
+	if err := m.Client.Get(ctx, client.ObjectKey{Name: val, Namespace: req.Namespace}, &cm); errors.IsNotFound(err) {
+		err := m.CreateConfigMap(ctx, val, req.Namespace, pod)
 		if err != nil {
-			return admission.Denied("FeatureFlagConfiguration not found")
+			m.Log.V(1).Info(fmt.Sprintf("failed to create config map %s error: %s", val, err.Error()))
+			return admission.Errored(http.StatusInternalServerError, err)
 		}
 	}
-	// TODO: this should be a short sha to avoid collisions
-	configName := name
-	// Create the agent configmap
-	m.Client.Delete(context.TODO(), &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      configName,
-			Namespace: req.Namespace,
-		},
-	}) // Delete the configmap if it exists
 
-	m.Log.V(1).Info(fmt.Sprintf("Creating configmap %s", configName))
-	if err := m.Client.Create(ctx, &corev1.ConfigMap{
+	if !CheckOwnerReference(pod, cm) {
+		reference := pod.OwnerReferences[0]
+		reference.Controller = False()
+		cm.OwnerReferences = append(cm.OwnerReferences, reference)
+		err := m.Client.Update(ctx, &cm)
+		if err != nil {
+			m.Log.V(1).Info(fmt.Sprintf("failed to update owner reference for %s error: %s", val, err.Error()))
+		}
+	}
+
+	marshaledPod, err := m.InjectSidecar(pod, val)
```

Comment:
We either should make all the object creation a function return or none of it, to stick to a single clear way of writing

Reply:
As discussed, we will stick with this at the moment